standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/model/MPackage.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.hadoop.hive.metastore.model;
+
+import org.apache.hadoop.hive.metastore.api.MetaException;
+
+public class MPackage {
+  private String name;
+  private MDatabase database;
+  private String owner;
+  private String header;
+  private String body;
+  private int createTime = (int)(System.currentTimeMillis() / 1000);
+  public static final int MAX_SOURCE_SIZE = 1073741823;
+
+  public MPackage() {}
+
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public MDatabase getDatabase() {
+    return database;
+  }
+
+  public void setDatabase(MDatabase database) {
+    this.database = database;
+  }
+
+  public String getOwner() {
+    return owner;
+  }
+
+  public void setOwner(String owner) {
+    this.owner = owner;
+  }
+
+  public String getHeader() {
+    return header;
+  }
+
+  public void setHeader(String header) throws MetaException {
+    if (header.length() > MAX_SOURCE_SIZE) {
```

Comment:
Does header and body need to be under MAX_SOURCE_SIZE individually or together? These checks do not prevent these two going over the limit together.

Reply:
This limit is per column. Each of those are stored in different columns.

Reply:
If MAX_SOURCE_SIZE applies to both individually, please name the constant to indicate that. Alternatively introduce another constant MAX_HEADER_SIZE = MAX_SOURCE_SIZE